mmf/models/visual_bert.py

```diff
@@ -416,123 +412,134 @@ def build(self):
             for p in self.model.bert.parameters():
                 p.requires_grad = False
 
-    def flatten(self, sample_list, to_be_flattened=None, to_be_flattened_dim=None):
-        if to_be_flattened is None:
-            to_be_flattened = {}
-        if to_be_flattened_dim is None:
-            to_be_flattened_dim = {}
+    def flatten(
+        self,
+        sample_list: Dict[str, Tensor],
+        to_be_flattened: List[str],
+        to_be_flattened_dim: List[str],
+    ) -> Dict[str, Tensor]:
         for key in to_be_flattened:
             # Make sure these keys are present or otherwise set these keys to None
-            sample_list[key] = getattr(sample_list, key, None)
             sample_list[key] = transform_to_batch_sequence(sample_list[key])
         for key in to_be_flattened_dim:
-            sample_list[key] = getattr(sample_list, key, None)
             sample_list[key] = transform_to_batch_sequence_dim(sample_list[key])
+        return sample_list
 
-        if sample_list.visual_embeddings_type is None:
-            if sample_list.image_mask is not None:
-                sample_list.visual_embeddings_type = torch.zeros_like(
-                    sample_list.image_mask
-                )
+    def add_post_flatten_params(
+        self, sample_list: Dict[str, Tensor]
+    ) -> Dict[str, Tensor]:
+        sample_list["visual_embeddings_type"] = torch.zeros_like(
+            sample_list["image_mask"]
+        )
 
-        if sample_list.image_mask is not None:
+        if sample_list["image_mask"] is not None:
```

Comment:
Since the previous `if sample_list.image_mask is not None:` is removed can we also remove this `if-else`?